server/src/client_api.rs

```diff
@@ -201,4 +252,170 @@ mod tests {
         assert_eq!(found_metric.no, 0);
         assert_eq!(found_metric.no, expected.no);
     }
+
+    fn cached_client_features() -> ClientFeatures {
+        ClientFeatures {
+            version: 2,
+            features: vec![
+                ClientFeature {
```

Comment:
I think we have a `Default` impl on ClientFeature. Entirely optional but it might make this huge blob easier to read